root/collection/CollectionIndex.js

```diff
@@ -0,0 +1,224 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import {withCatalystContext} from '../context';
+import AreaList from '../components/AreaList';
+import ArtistList from '../components/ArtistList';
+import EventList from '../components/EventList';
+import FormRow from '../components/FormRow';
+import FormSubmit from '../components/FormSubmit';
+import InstrumentList from '../components/InstrumentList';
+import LabelList from '../components/LabelList';
+import PlaceList from '../components/PlaceList';
+import RecordingList from '../components/RecordingList';
+import ReleaseGroupList from '../components/ReleaseGroupList';
+import ReleaseList from '../components/ReleaseList';
+import SeriesList from '../components/SeriesList';
+import WorkList from '../components/WorkList';
+import PaginatedResults from '../components/PaginatedResults';
+import expand2react from '../static/scripts/common/i18n/expand2react';
+import {formatPluralEntityTypeName}
+  from '../static/scripts/common/utility/formatEntityTypeName';
+
+import CollectionLayout from './CollectionLayout';
+
+type PropsForEntity<T: CoreEntityT> = {
+  +$c: CatalystContextT,
+  +collection: CollectionT,
+  +collectionEntityType: $ElementType<T, 'entityType'>,
+  +entities: $ReadOnlyArray<T>,
+  +order: string,
+  +pager: PagerT,
+};
+
+type Props =
+  | PropsForEntity<AreaT>
+  | PropsForEntity<ArtistT>
+  | PropsForEntity<EventT>
+  | PropsForEntity<InstrumentT>
+  | PropsForEntity<LabelT>
+  | PropsForEntity<PlaceT>
+  | PropsForEntity<RecordingT>
+  | PropsForEntity<ReleaseGroupT>
+  | PropsForEntity<ReleaseT>
+  | PropsForEntity<SeriesT>
+  | PropsForEntity<WorkT>
+  ;
+
+const listPicker = (props: Props, ownCollection: boolean) => {
+  switch (props.collectionEntityType) {
+    case 'area':
+      return (
+        <AreaList
+          areas={props.entities}
+          checkboxes={ownCollection ? 'remove' : ''}
+          order={props.order}
+          sortable
```

Comment:
Is there a way to just specify `checkboxes={ownCollection ? 'remove' : ''}`, `order={props.order}
` and `sortable` once and then pass it to all? Possibly also `entities={props.entities}`, if we choose to generalize that on the lists.

Reply:
If you put them in an object fist, you can use JSX spread syntax:

```JSX
const sharedProps = {
  checkboxes: ownCollection ? 'remove' : '',
  order: props.order,
  sortable: true,
};

// ...

<AreaList
  areas={props.entities}
  {...sharedProps}
/>
```

For `areas={props.entities}` you can either generalize all the prop names to `entities` as you mentioned, or look up the plural name in entities.json.